libjob_queue/src/lsf_driver.c

```diff
@@ -763,8 +763,25 @@ void lsf_driver_kill_job(void * __driver , void * __job) {
 
         free( argv[1] );
         free( argv );
-      } else if (driver->submit_method == LSF_SUBMIT_LOCAL_SHELL)
+      } else if (driver->submit_method == LSF_SUBMIT_LOCAL_SHELL) {
         util_spawn_blocking(driver->bkill_cmd, 1, (const char **) &job->lsf_jobnr_char, NULL, NULL);
+      }
+
+      if (blacklist) {
```

Comment:
If the blacklisting should be part of the kill the assembly of hostnames should be performed *before* the kill - otherwise I guess we could risk that the LSF server had started to tear down the hostname information.